content/blog/ggplot2-3-5-0-axes/index.Rmd

```diff
@@ -0,0 +1,223 @@
+#|  the top break. The x-axis line starts at the most left break and ends at
+#|  the most right break.
+ggplot(mpg, aes(displ, hwy)) +
+  geom_point() +
+  guides(
+    x = guide_axis(cap = "both"), # Cap both ends
+    y = guide_axis(cap = "upper") # Cap the upper end
+  ) +
+  theme(axis.line = element_line())
+```
+
+## Logarithmic axes
+
+A new axis for displaying logarithmic (and related) scales has been added: `guide_axis_logticks()`.
+This axis draws three types of tick marks at log10-spaced positions.
+The ticks positions are placed in the original, untransformed data-space, so the axis plays well with scale- and coord-transformations.
+To accommodate a series of logarithmic-like transformations, such as `scales::transform_pseudo_log()` or `scales::transform_asinh()`, scales that include 0 in their limits have the ticks mirrored around 0.
+
+```{r log_axes}
+#| fig.alt: >
+#|  A line plot showing a negatively sloped line with a reversed
+#|  log10-transformation on the y-axis and inverse hyberbolic sine transformation
+#|  on the x-axis. Large ticks appears at multiples of 10, 
+#|  medium ticks at multiples of 5 and small ticks at multiples of 1.
+
+r <- seq(0.001, 0.999, length.out = 100)
+df <- data.frame(
+  x = qcauchy(r),
+  y = qlnorm(r)
+)
+
+p <- ggplot(df, aes(x, y)) +
+  geom_line() +
+  coord_trans(y = "reverse") +
+  scale_y_continuous(
+    transform = "log10",
+    breaks = c(0.1, 1, 10),
+    guide = guide_axis_logticks(long = 2, mid = 1, short = 0.5)
+  ) +
+  scale_x_continuous(
+    transform = "asinh",
+    breaks = c(-100, -10, -1, 0, 1, 10, 100),
+    guide = "axis_logticks"
+  )
+p
+```
+
+The log-ticks axis supersedes the earlier `annotation_logticks()` function.
+Because it is implemented as an axis, it has minimal fuss with the placement of labels and is immune to the clipping options in the coord.
+To mirror `annotation_logticks()` more closely, you can set a negative tick length in the theme.
+
+```{r log_ticks_inward}
+#| fig.alt: >
+#|  The same plot as above, but the tick marks now point inwards.
+
+p + theme(axis.ticks.length = unit(-2.25, "pt"))
+```
+
+## Stacked axes
+
+The last new axis is technically not an axis, but a way to combine axis.
+`guide_axis_stack()` can take multiple other axes and combine them by placing them next to oneanother.
+On its own, the usefulness of stacking axes is pretty limited.
+However, when extensions start defining custom position guides, it is an easy way to mix-and-match axes from different extensions.
+The first axis is placed next to the panel and subsequent axes are placed further away from the panel.
+Axes, like legends, have acquired a `theme` argument that can be used to customise the display of individual axes.
+
+```{r stack_axis}
+#| fig.alt: >
+#|  Scatterplot of engine displacement versus highway miles per 
+#|  gallon. The x-axis resembles railroad tracks with labels in the middle and
+#|  an additional line beneath.
+
+ggplot(mpg, aes(displ, hwy)) +
```

Comment:
Do you have a more compelling example to show off here? I don't get why people would want this.

Reply:
I think this is more compelling once guide extensions arrives - i.e. you could stack a discrete axis with a table giving different traits for the various levels... Or you could have a continuous axis with qualitative axis below to explain what the limits of the scale means (bad <-> good)... But I don't know off-hand of a compelling example that can be made with just the standard axes available from ggplot2

Reply:
I also don't think there is a compelling example to make with just the axes in ggplot2.

To give an example with some WIP guide extension, you could imagine a guide that draws labelled brackets at some interval, which would be a separate guide, but can be combined with the standard axis.

![image](https://github.com/tidyverse/tidyverse.org/assets/49372158/a9801809-19d7-4f05-8df4-034025ca4d88)

Reply:
In that case, I think you might just mention this without an example, highlighting that it's an important building block for future work.